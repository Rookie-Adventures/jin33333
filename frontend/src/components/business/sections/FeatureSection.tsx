import { Box, Grid, Typography } from '@mui/material';

import type { FC } from 'react';

import { AnimatedCard } from '@/components/common/cards/AnimatedCard';
import { SectionLayout } from '@/components/common/layout/SectionLayout';

import { features } from './config/features';

// 提取特性卡片组件
const FeatureCard: FC<{
  title: string;
  description: string;
  Icon: FC<{ sx?: object }>;
  index: number;
  delay: number;
}> = ({ title, description, Icon, index, delay }) => (
  <AnimatedCard delay={delay}>
    <Box
      sx={{
        p: 3,
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        textAlign: 'center',
      }}
    >
      <Box
        sx={{
          mb: 2,
          p: 1.5,
          borderRadius: '50%',
          background: 'linear-gradient(135deg, #24243e 0%, #302b63 100%)',
          color: 'white',
        }}
      >
        <Icon data-testid={`feature-icon-${index}`} sx={{ fontSize: 40 }} />
      </Box>
      <Typography
        variant="h5"
        component="h3"
        gutterBottom
        sx={{ fontWeight: 'bold', mb: 1 }}
      >
        {title}
      </Typography>
      <Typography variant="body1" color="text.secondary" sx={{ flexGrow: 1 }}>
        {description}
      </Typography>
    </Box>
  </AnimatedCard>
);

const FeatureSection: FC = () => (
  <SectionLayout
    title="核心功能"
    subtitle="我们提供全面的 AI 模型服务，让您的应用更智能"
  >
    <Grid container spacing={3} justifyContent="center">
      {features.map((feature, index) => (
        <Grid
          item
          key={feature.title}
          xs={12}
          sm={6}
          md={3}
          sx={{
            display: 'flex',
            justifyContent: 'center',
          }}
        >
          <FeatureCard
            title={feature.title}
            description={feature.description}
            Icon={feature.icon}
            index={index}
            delay={feature.delay}
          />
        </Grid>
      ))}
    </Grid>
  </SectionLayout>
);

export default FeatureSection;
