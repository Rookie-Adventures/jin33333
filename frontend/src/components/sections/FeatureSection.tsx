import React from 'react';
import { Grid, Card, CardContent, Typography, Box, useTheme } from '@mui/material';
import { 
  Speed, 
  Security, 
  CloudSync,
  Psychology,
  Token,
  Autorenew,
  DataObject,
  Settings
} from '@mui/icons-material';
import { motion } from 'framer-motion';

const features = [
  {
    icon: <Speed />,
    title: '流式响应体验',
    description: '支持流式输出，实现即时响应，提供流畅的对话体验',
  },
  {
    icon: <CloudSync />,
    title: '智能代理访问',
    description: '自动判断用户地区，智能使用代理，确保服务稳定可用',
  },
  {
    icon: <Psychology />,
    title: '智能上下文',
    description: '先进的上下文管理，支持多轮对话，提供连贯的交互体验',
  },
  {
    icon: <Token />,
    title: 'Token 优化',
    description: '智能的 Token 使用优化，平衡体验与成本',
  },
  {
    icon: <Security />,
    title: '安全可靠',
    description: '完善的安全机制，保护用户数据和API密钥安全',
  },
  {
    icon: <Autorenew />,
    title: '自动重试',
    description: '智能错误处理，自动重试机制，确保服务稳定性',
  },
  {
    icon: <DataObject />,
    title: '统一API',
    description: '提供统一的API接口，支持多种模型调用',
  },
  {
    icon: <Settings />,
    title: '高度定制',
    description: '支持个性化配置，满足不同场景需求',
  },
];

const FeatureSection: React.FC = () => {
  const theme = useTheme();

  return (
    <Box sx={{ 
      py: { xs: 4, md: 6 },
      px: { xs: 0.5, sm: 1 },
      background: 'linear-gradient(180deg, rgba(255,255,255,0) 0%, rgba(245,247,250,0.5) 100%)',
      width: '100%'
    }}>
      <Typography
        variant="h3"
        component="h2"
        align="center"
        gutterBottom
        sx={{ 
          mb: 2,
          fontSize: { xs: '2rem', md: '3rem' },
          fontWeight: 'bold',
          background: 'linear-gradient(135deg, #24243e 0%, #302b63 100%)',
          WebkitBackgroundClip: 'text',
          WebkitTextFillColor: 'transparent',
        }}
      >
        核心特性
      </Typography>
      <Typography
        variant="h6"
        component="p"
        align="center"
        color="text.secondary"
        sx={{ mb: { xs: 3, md: 4 } }}
      >
        为您提供最优质的 HuggingFace 模型调用服务
      </Typography>
      <Grid container spacing={0.5}>
        {features.map((feature, index) => (
          <Grid item xs={12} sm={6} md={3} key={index} sx={{ pl: { xs: 0.5, sm: 1 } }}>
            <motion.div
              initial={{ opacity: 0, y: 20 }}
              animate={{ opacity: 1, y: 0 }}
              transition={{ duration: 0.5, delay: index * 0.1 }}
              style={{ height: '100%' }}
            >
              <Card
                elevation={0}
                sx={{
                  height: '100%',
                  display: 'flex',
                  flexDirection: 'column',
                  transition: 'all 0.3s ease-in-out',
                  background: 'rgba(255, 255, 255, 0.8)',
                  backdropFilter: 'blur(10px)',
                  border: '1px solid rgba(255, 255, 255, 0.3)',
                  borderRadius: 1,
                  '&:hover': {
                    transform: 'translateY(-8px)',
                    boxShadow: theme.shadows[8],
                    background: 'rgba(255, 255, 255, 0.95)',
                  },
                }}
              >
                <CardContent sx={{ p: 1.5 }}>
                  <Box
                    sx={{
                      display: 'flex',
                      justifyContent: 'center',
                      mb: 1,
                      color: 'primary.main',
                      '& > svg': { 
                        fontSize: { xs: 32, md: 40 },
                        filter: 'drop-shadow(0px 2px 4px rgba(0,0,0,0.1))',
                      },
                    }}
                  >
                    {feature.icon}
                  </Box>
                  <Typography
                    gutterBottom
                    variant="h6"
                    component="h3"
                    align="center"
                    sx={{ 
                      fontWeight: 'bold',
                      fontSize: { xs: '1rem', md: '1.15rem' },
                      mb: 0.5,
                    }}
                  >
                    {feature.title}
                  </Typography>
                  <Typography
                    variant="body2"
                    color="text.secondary"
                    align="center"
                    sx={{
                      fontSize: { xs: '0.875rem', md: '0.9rem' },
                      lineHeight: 1.5,
                    }}
                  >
                    {feature.description}
                  </Typography>
                </CardContent>
              </Card>
            </motion.div>
          </Grid>
        ))}
      </Grid>
    </Box>
  );
};

export default FeatureSection; 