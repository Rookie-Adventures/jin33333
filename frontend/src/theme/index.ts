import { createTheme as createMuiTheme } from '@mui/material/styles';
import type { ThemeOptions } from '@mui/material/styles';

export const defaultTheme: ThemeOptions = {
  palette: {
    mode: 'light' as const,
    primary: {
      main: '#1976d2',
      light: '#42a5f5',
      dark: '#1565c0',
      contrastText: '#ffffff'
    },
    secondary: {
      main: '#9c27b0',
      light: '#ba68c8',
      dark: '#7b1fa2',
      contrastText: '#ffffff'
    },
    error: {
      main: '#d32f2f',
      light: '#ef5350',
      dark: '#c62828',
      contrastText: '#ffffff'
    },
    warning: {
      main: '#ed6c02',
      light: '#ff9800',
      dark: '#e65100',
      contrastText: '#ffffff'
    },
    info: {
      main: '#0288d1',
      light: '#03a9f4',
      dark: '#01579b',
      contrastText: '#ffffff'
    },
    success: {
      main: '#2e7d32',
      light: '#4caf50',
      dark: '#1b5e20',
      contrastText: '#ffffff'
    },
    background: {
      default: '#ffffff',
      paper: '#f5f5f5'
    },
    text: {
      primary: 'rgba(0, 0, 0, 0.87)',
      secondary: 'rgba(0, 0, 0, 0.6)'
    }
  },
  typography: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    h1: {
      fontSize: '2.5rem',
      fontWeight: 500
    },
    h2: {
      fontSize: '2rem',
      fontWeight: 500
    },
    h3: {
      fontSize: '1.75rem',
      fontWeight: 500
    },
    h4: {
      fontSize: '1.5rem',
      fontWeight: 500
    },
    h5: {
      fontSize: '1.25rem',
      fontWeight: 500
    },
    h6: {
      fontSize: '1rem',
      fontWeight: 500
    },
    body1: {
      fontSize: '1rem',
      lineHeight: 1.5
    },
    body2: {
      fontSize: '0.875rem',
      lineHeight: 1.43
    },
    button: {
      fontSize: '0.875rem',
      textTransform: 'uppercase',
      fontWeight: 500
    }
  },
  spacing: 8,
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 960,
      lg: 1280,
      xl: 1920
    }
  },
  transitions: {
    easing: {
      easeInOut: 'cubic-bezier(0.4, 0, 0.2, 1)',
      easeOut: 'cubic-bezier(0.0, 0, 0.2, 1)',
      easeIn: 'cubic-bezier(0.4, 0, 1, 1)',
      sharp: 'cubic-bezier(0.4, 0, 0.6, 1)'
    },
    duration: {
      shortest: 150,
      shorter: 200,
      short: 250,
      standard: 300,
      complex: 375,
      enteringScreen: 225,
      leavingScreen: 195
    }
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: 'none'
        } as const
      }
    },
    MuiTextField: {
      defaultProps: {
        variant: 'outlined'
      }
    },
    MuiCard: {
      styleOverrides: {
        root: {
          borderRadius: 8
        }
      }
    }
  }
};

export const darkTheme: ThemeOptions = {
  ...defaultTheme,
  palette: {
    ...defaultTheme.palette,
    mode: 'dark' as const,
    background: {
      default: '#121212',
      paper: '#1e1e1e'
    },
    text: {
      primary: '#ffffff',
      secondary: 'rgba(255, 255, 255, 0.7)'
    },
    action: {
      active: '#fff',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.16)',
      disabled: 'rgba(255, 255, 255, 0.3)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)'
    }
  }
};

export const createTheme = (isDark: boolean, customOptions: Partial<ThemeOptions> = {}): ThemeOptions => {
  const baseTheme = isDark ? darkTheme : defaultTheme;
  return createMuiTheme(baseTheme, customOptions);
};

// 导出默认主题实例
export const theme = createTheme(false); 