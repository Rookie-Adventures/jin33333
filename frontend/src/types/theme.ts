export type ThemeMode = 'light' | 'dark';

export interface ThemeCustomizations {
  primaryColor: string;
}

export interface ThemeState {
  mode: ThemeMode;
  customizations: ThemeCustomizations;
} 