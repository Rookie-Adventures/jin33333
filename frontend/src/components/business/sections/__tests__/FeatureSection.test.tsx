import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from '@mui/material/styles';
import { useMediaQuery } from '@mui/material';
import { vi } from 'vitest';

import FeatureSection from '../FeatureSection';
import { theme } from '@/theme';

// Mock useMediaQuery
vi.mock('@mui/material', async () => {
  const actual = await vi.importActual('@mui/material');
  return {
    ...actual as any,
    useMediaQuery: vi.fn()
  };
});

describe('FeatureSection', () => {
  const renderFeatureSection = () => {
    return render(
      <ThemeProvider theme={theme(false)}>
        <FeatureSection />
      </ThemeProvider>
    );
  };

  beforeEach(() => {
    // 默认设置为桌面视图
    vi.mocked(useMediaQuery).mockReturnValue(false);
  });

  it('应该正确渲染标题和描述', () => {
    renderFeatureSection();
    
    expect(screen.getByText('核心功能')).toBeInTheDocument();
    expect(screen.getByText(/我们提供全面的 AI 模型服务/)).toBeInTheDocument();
  });

  it('应该正确渲染所有功能卡片', () => {
    renderFeatureSection();
    
    // 验证功能卡片
    expect(screen.getByText('流式响应')).toBeInTheDocument();
    expect(screen.getByText('智能上下文')).toBeInTheDocument();
    expect(screen.getByText('多模型支持')).toBeInTheDocument();
    expect(screen.getByText('安全可控')).toBeInTheDocument();
  });

  it('应该正确显示功能描述', () => {
    renderFeatureSection();
    
    expect(screen.getByText(/支持流式输出，实时响应/)).toBeInTheDocument();
    expect(screen.getByText(/智能管理对话上下文/)).toBeInTheDocument();
    expect(screen.getByText(/支持多种主流模型/)).toBeInTheDocument();
    expect(screen.getByText(/数据安全有保障/)).toBeInTheDocument();
  });

  it('应该在移动端正确响应', () => {
    // 模拟移动端视图
    vi.mocked(useMediaQuery).mockReturnValue(true);
    
    renderFeatureSection();
    
    const cards = screen.getAllByRole('article');
    cards.forEach(card => {
      expect(card).toHaveStyle({
        margin: '0 auto',
        maxWidth: '350px'
      });
    });
  });

  it('应该正确应用动画效果', () => {
    renderFeatureSection();
    
    const cards = screen.getAllByRole('article');
    cards.forEach(card => {
      expect(card).toHaveStyle({
        opacity: 1,
        transform: 'translateY(0)'
      });
    });
  });

  it('应该正确显示图标', () => {
    renderFeatureSection();
    
    const icons = screen.getAllByTestId(/feature-icon/);
    expect(icons).toHaveLength(4);
    icons.forEach(icon => {
      expect(icon).toBeInTheDocument();
    });
  });

  // 添加快照测试
  it('应该匹配桌面端快照', () => {
    const { container } = renderFeatureSection();
    expect(container).toMatchSnapshot();
  });

  it('应该匹配移动端快照', () => {
    vi.mocked(useMediaQuery).mockReturnValue(true);
    const { container } = renderFeatureSection();
    expect(container).toMatchSnapshot();
  });
}); 