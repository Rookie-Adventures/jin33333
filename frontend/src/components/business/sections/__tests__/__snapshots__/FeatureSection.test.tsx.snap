// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeatureSection > 应该匹配桌面端快照 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
  >
    <div
      aria-label="核心功能"
      class="MuiBox-root css-1hggc9u"
      role="main"
    >
      <h2
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter MuiTypography-gutterBottom css-k6sdrn-MuiTypography-root"
      >
        核心功能
      </h2>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-10jcxlz-MuiTypography-root"
      >
        我们提供全面的 AI 模型服务，让您的应用更智能
      </h6>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-8wqzlf-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-5jkmth-MuiGrid-root"
        >
          <div
            style="width: 100%; height: 100%; opacity: 0; transform: translateY(20px);"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1p7m8ke-MuiPaper-root-MuiCard-root"
              role="article"
            >
              <div
                class="MuiBox-root css-14d20i8"
              >
                <div
                  class="MuiBox-root css-16metcc"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ha5wk2-MuiSvgIcon-root"
                    data-testid="feature-icon-0"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19.03 3.56c-1.67-1.39-3.74-2.3-6.03-2.51v2.01c1.73.19 3.31.88 4.61 1.92zM11 3.06V1.05c-2.29.2-4.36 1.12-6.03 2.51l1.42 1.42C7.69 3.94 9.27 3.25 11 3.06M4.98 6.39 3.56 4.97C2.17 6.64 1.26 8.71 1.05 11h2.01c.19-1.73.88-3.31 1.92-4.61M20.94 11h2.01c-.21-2.29-1.12-4.36-2.51-6.03l-1.42 1.42c1.04 1.3 1.73 2.88 1.92 4.61M7 12l3.44 1.56L12 17l1.56-3.44L17 12l-3.44-1.56L12 7l-1.56 3.44z"
                    />
                    <path
                      d="M12 21c-3.11 0-5.85-1.59-7.46-4H7v-2H1v6h2v-2.7c1.99 2.84 5.27 4.7 9 4.7 4.87 0 9-3.17 10.44-7.56l-1.96-.45C19.25 18.48 15.92 21 12 21"
                    />
                  </svg>
                </div>
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1ndmryj-MuiTypography-root"
                >
                  流式响应
                </h3>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1leyb9s-MuiTypography-root"
                >
                  支持流式输出，实时响应，提供更自然的对话体验
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-5jkmth-MuiGrid-root"
        >
          <div
            style="width: 100%; height: 100%; opacity: 0; transform: translateY(20px);"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1p7m8ke-MuiPaper-root-MuiCard-root"
              role="article"
            >
              <div
                class="MuiBox-root css-14d20i8"
              >
                <div
                  class="MuiBox-root css-16metcc"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ha5wk2-MuiSvgIcon-root"
                    data-testid="feature-icon-1"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 8.57c-.79 0-1.43.64-1.43 1.43s.64 1.43 1.43 1.43 1.43-.64 1.43-1.43-.64-1.43-1.43-1.43"
                    />
                    <path
                      d="M13 3C9.25 3 6.2 5.94 6.02 9.64L4.1 12.2c-.25.33-.01.8.4.8H6v3c0 1.1.9 2 2 2h1v3h7v-4.68c2.36-1.12 4-3.53 4-6.32 0-3.87-3.13-7-7-7m3 7c0 .13-.01.26-.02.39l.83.66c.08.06.1.16.05.25l-.8 1.39c-.05.09-.16.12-.24.09l-.99-.4c-.21.16-.43.29-.67.39L14 13.83c-.01.1-.1.17-.2.17h-1.6c-.1 0-.18-.07-.2-.17l-.15-1.06c-.25-.1-.47-.23-.68-.39l-.99.4c-.09.03-.2 0-.25-.09l-.8-1.39c-.05-.08-.03-.19.05-.25l.84-.66c-.01-.13-.02-.26-.02-.39s.02-.27.04-.39l-.85-.66c-.08-.06-.1-.16-.05-.26l.8-1.38c.05-.09.15-.12.24-.09l1 .4c.2-.15.43-.29.67-.39L12 6.17c.02-.1.1-.17.2-.17h1.6c.1 0 .18.07.2.17l.15 1.06c.24.1.46.23.67.39l1-.4c.09-.03.2 0 .24.09l.8 1.38c.05.09.03.2-.05.26l-.85.66c.03.12.04.25.04.39"
                    />
                  </svg>
                </div>
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1ndmryj-MuiTypography-root"
                >
                  智能上下文
                </h3>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1leyb9s-MuiTypography-root"
                >
                  智能管理对话上下文，实现连贯的多轮对话
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-5jkmth-MuiGrid-root"
        >
          <div
            style="width: 100%; height: 100%; opacity: 0; transform: translateY(20px);"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1p7m8ke-MuiPaper-root-MuiCard-root"
              role="article"
            >
              <div
                class="MuiBox-root css-14d20i8"
              >
                <div
                  class="MuiBox-root css-16metcc"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ha5wk2-MuiSvgIcon-root"
                    data-testid="feature-icon-2"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M4 8h4V4H4zm6 12h4v-4h-4zm-6 0h4v-4H4zm0-6h4v-4H4zm6 0h4v-4h-4zm6-10v4h4V4zm-6 4h4V4h-4zm6 6h4v-4h-4zm0 6h4v-4h-4z"
                    />
                  </svg>
                </div>
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1ndmryj-MuiTypography-root"
                >
                  多模型支持
                </h3>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1leyb9s-MuiTypography-root"
                >
                  支持多种主流模型，满足不同场景需求
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-5jkmth-MuiGrid-root"
        >
          <div
            style="width: 100%; height: 100%; opacity: 0; transform: translateY(20px);"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-1p7m8ke-MuiPaper-root-MuiCard-root"
              role="article"
            >
              <div
                class="MuiBox-root css-14d20i8"
              >
                <div
                  class="MuiBox-root css-16metcc"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ha5wk2-MuiSvgIcon-root"
                    data-testid="feature-icon-3"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 1 3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11z"
                    />
                  </svg>
                </div>
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1ndmryj-MuiTypography-root"
                >
                  安全可控
                </h3>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1leyb9s-MuiTypography-root"
                >
                  数据安全有保障，支持自定义敏感词过滤
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FeatureSection > 应该匹配移动端快照 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-cnms5y-MuiContainer-root"
  >
    <div
      aria-label="核心功能"
      class="MuiBox-root css-1hggc9u"
      role="main"
    >
      <h2
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter MuiTypography-gutterBottom css-k6sdrn-MuiTypography-root"
      >
        核心功能
      </h2>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-10jcxlz-MuiTypography-root"
      >
        我们提供全面的 AI 模型服务，让您的应用更智能
      </h6>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-8wqzlf-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-5jkmth-MuiGrid-root"
        >
          <div
            style="width: 100%; height: 100%; opacity: 0; transform: translateY(20px);"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-19ohs2l-MuiPaper-root-MuiCard-root"
              role="article"
            >
              <div
                class="MuiBox-root css-14d20i8"
              >
                <div
                  class="MuiBox-root css-16metcc"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ha5wk2-MuiSvgIcon-root"
                    data-testid="feature-icon-0"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19.03 3.56c-1.67-1.39-3.74-2.3-6.03-2.51v2.01c1.73.19 3.31.88 4.61 1.92zM11 3.06V1.05c-2.29.2-4.36 1.12-6.03 2.51l1.42 1.42C7.69 3.94 9.27 3.25 11 3.06M4.98 6.39 3.56 4.97C2.17 6.64 1.26 8.71 1.05 11h2.01c.19-1.73.88-3.31 1.92-4.61M20.94 11h2.01c-.21-2.29-1.12-4.36-2.51-6.03l-1.42 1.42c1.04 1.3 1.73 2.88 1.92 4.61M7 12l3.44 1.56L12 17l1.56-3.44L17 12l-3.44-1.56L12 7l-1.56 3.44z"
                    />
                    <path
                      d="M12 21c-3.11 0-5.85-1.59-7.46-4H7v-2H1v6h2v-2.7c1.99 2.84 5.27 4.7 9 4.7 4.87 0 9-3.17 10.44-7.56l-1.96-.45C19.25 18.48 15.92 21 12 21"
                    />
                  </svg>
                </div>
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1ndmryj-MuiTypography-root"
                >
                  流式响应
                </h3>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1leyb9s-MuiTypography-root"
                >
                  支持流式输出，实时响应，提供更自然的对话体验
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-5jkmth-MuiGrid-root"
        >
          <div
            style="width: 100%; height: 100%; opacity: 0; transform: translateY(20px);"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-19ohs2l-MuiPaper-root-MuiCard-root"
              role="article"
            >
              <div
                class="MuiBox-root css-14d20i8"
              >
                <div
                  class="MuiBox-root css-16metcc"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ha5wk2-MuiSvgIcon-root"
                    data-testid="feature-icon-1"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 8.57c-.79 0-1.43.64-1.43 1.43s.64 1.43 1.43 1.43 1.43-.64 1.43-1.43-.64-1.43-1.43-1.43"
                    />
                    <path
                      d="M13 3C9.25 3 6.2 5.94 6.02 9.64L4.1 12.2c-.25.33-.01.8.4.8H6v3c0 1.1.9 2 2 2h1v3h7v-4.68c2.36-1.12 4-3.53 4-6.32 0-3.87-3.13-7-7-7m3 7c0 .13-.01.26-.02.39l.83.66c.08.06.1.16.05.25l-.8 1.39c-.05.09-.16.12-.24.09l-.99-.4c-.21.16-.43.29-.67.39L14 13.83c-.01.1-.1.17-.2.17h-1.6c-.1 0-.18-.07-.2-.17l-.15-1.06c-.25-.1-.47-.23-.68-.39l-.99.4c-.09.03-.2 0-.25-.09l-.8-1.39c-.05-.08-.03-.19.05-.25l.84-.66c-.01-.13-.02-.26-.02-.39s.02-.27.04-.39l-.85-.66c-.08-.06-.1-.16-.05-.26l.8-1.38c.05-.09.15-.12.24-.09l1 .4c.2-.15.43-.29.67-.39L12 6.17c.02-.1.1-.17.2-.17h1.6c.1 0 .18.07.2.17l.15 1.06c.24.1.46.23.67.39l1-.4c.09-.03.2 0 .24.09l.8 1.38c.05.09.03.2-.05.26l-.85.66c.03.12.04.25.04.39"
                    />
                  </svg>
                </div>
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1ndmryj-MuiTypography-root"
                >
                  智能上下文
                </h3>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1leyb9s-MuiTypography-root"
                >
                  智能管理对话上下文，实现连贯的多轮对话
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-5jkmth-MuiGrid-root"
        >
          <div
            style="width: 100%; height: 100%; opacity: 0; transform: translateY(20px);"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-19ohs2l-MuiPaper-root-MuiCard-root"
              role="article"
            >
              <div
                class="MuiBox-root css-14d20i8"
              >
                <div
                  class="MuiBox-root css-16metcc"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ha5wk2-MuiSvgIcon-root"
                    data-testid="feature-icon-2"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M4 8h4V4H4zm6 12h4v-4h-4zm-6 0h4v-4H4zm0-6h4v-4H4zm6 0h4v-4h-4zm6-10v4h4V4zm-6 4h4V4h-4zm6 6h4v-4h-4zm0 6h4v-4h-4z"
                    />
                  </svg>
                </div>
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1ndmryj-MuiTypography-root"
                >
                  多模型支持
                </h3>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1leyb9s-MuiTypography-root"
                >
                  支持多种主流模型，满足不同场景需求
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3 css-5jkmth-MuiGrid-root"
        >
          <div
            style="width: 100%; height: 100%; opacity: 0; transform: translateY(20px);"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root css-19ohs2l-MuiPaper-root-MuiCard-root"
              role="article"
            >
              <div
                class="MuiBox-root css-14d20i8"
              >
                <div
                  class="MuiBox-root css-16metcc"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ha5wk2-MuiSvgIcon-root"
                    data-testid="feature-icon-3"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 1 3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11z"
                    />
                  </svg>
                </div>
                <h3
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1ndmryj-MuiTypography-root"
                >
                  安全可控
                </h3>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1leyb9s-MuiTypography-root"
                >
                  数据安全有保障，支持自定义敏感词过滤
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
